### DISCUSSION TITLE
Automating Authentication for data

### DISCUSSION THREAD

**Discussion by shahid430** [grafana/grafana-infinity-datasource#500] [Q&A]
We configured infinity plugin for our application but there is no option for access key & secret key to be applied. We are using "X-Auth_token" to authenticate but the token expiry is 24 hours only, after which we need to reset the header with new token.

Could someone please provide the configuration file for secret & access key to get rid of manually entering the X-Auth-Token everyday & steps how to do it.

**Comment by shahid430** [comment]
@yesoreyeram could you please respond to the query, would be appreciated.

**Reply by Usama420-git** [reply]
https://github.com/yesoreyeram/grafana-infinity-datasource/discussions/260 did you read this?

**Reply by shahid430** [reply]
Yes i read this but it is not working for me.
What i need to pass my Huawei Access & Secret key to authenticate which never changes but i tried that in custom header but it didn't work.

**Reply by shahid430** [reply]
Let me explain my scenario hope you could solve my issue:
I am using infinity plugin inside grafana to fetch the data from huawei cloud using api's. To do so, i need to authenticate infinity plugin to fetch the data. So to get this, i am using huawei token to authenticate & i am getting the data but every 24 hours the token expires. Instead of using token i want to use Access Key & Secret Key to get rid of the manual process but it is not working.
Now we have both ways:
1. Authenticate via token only but need to be automated but i want to know the process how to do it though setup my yaml file inside /conf/provsioning/datasources/sample.yaml file & i am inserting Header Key Name(X-Auth-Token) & Header Key Value(Bearer Token Value). Now to use these headers i need to insert IAM API key with Domain Name & Password which i don't know how to pass it. 
2. Second way is to use IAM API with Huawei Access Key & Secret Key.

Please suggest the best option & procedure how to do it. Much Appreciated.

**Comment by shahid430** [comment]
Still no update on the issue. Shall i consider that this is no alternative solution to this or no body knows how to do it.

**Reply by yesoreyeram** [reply]
Out of the box, refresh token is only supported in oauth2 credentials type. 
But this is frequently requested features and so this being considered for future versions. 

Currently it is not possible out of the box. But the alternative is to use some cron jobs to call the grafana datasource api.

**Reply by shahid430** [reply]
Thank you for the update, we tried cronjob but it is not working , is there any file in the repository where we can define the credentials and call it via cronjob.
